src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/ServiceMatch.cs

```diff
@@ -247,18 +247,30 @@ public ExternalService(string typeName, string assemblyName, string assemblyPubl
 
             public string AssemblyPublicKey { get; }
 
-            public bool IsMatch(Type serviceType)
+            public bool IsMatch(Type serviceType, Type implementationFactory)
```

Comment:
can you pass the ServiceDescriptor itself in here instead? That'd be a cleaner API. Then you can have the logic to check its ImplementationType first, then fallback to ImplementationFactory, etc.